pkg/backend/display/rows.go

```diff
@@ -215,31 +217,16 @@ func (data *resourceRowData) recordDiagEventPayload(payload engine.DiagEventPayl
 	}
 }
 
-func (data *resourceRowData) RecordPolicyViolationEvent(event engine.Event) {
-
-	//
-	// NOTE: The display code only understands DiagEvents. We convert the policy violation events
-	// accordingly so they can be displayed.
-	//
+// PolicyInfo returns the PolicyInfo object associated with the resourceRowData.
+func (data *resourceRowData) PolicyInfo() []engine.PolicyViolationEventPayload {
+	return data.policyInfo
+}
 
+// RecordPolicyViolationEvent records a policy event with the resourceRowData.
+func (data *resourceRowData) RecordPolicyViolationEvent(event engine.Event) {
+	contract.Assert(event.Type == engine.PolicyViolationEvent)
```

Comment:
nit: assert really isn't needed.  it will just fire when ths is incorrect.  but the line below will also fire in that case.